modules/evm/src/lib.rs

```diff
@@ -678,11 +678,18 @@ pub mod module {
 
 impl<T: Config> Pallet<T> {
 	/// Process queued events
-	pub fn process_queued_events() {
+	pub fn process_queued_events() -> DispatchResult {
 		for event in Self::queued_events() {
+
+			if let Event::<T>::ContractSelfdestructed(contract, caller) = event {
```

Comment:
If contract removal fails the QueuedEvents will never be killed leaving events hanging.

Reply:
utilise `.drain(..)` to ensure vec is purged after every extrinsic - https://docs.substrate.io/rustdocs/latest/sp_std/vec/struct.Vec.html#method.drain

TODO: Add tests for failure case to ensure vec is cleared.